tests/unit/test_token_verifier.py

```diff
@@ -0,0 +1,416 @@
+        issuer = "https://example.com"
+        audience = "test-client"
+        
+        verifier = JWKSTokenVerifier(jwks_endpoint, issuer, audience)
+        
+        # Mock JWT decode to raise an exception for missing claims
+        with patch('jwt.decode', side_effect=jwt.MissingRequiredClaimError("Missing required claim: exp")):
+            result = verifier._decode_and_validate_jwt("mock.jwt.token", "mock-public-key")
+            
+            assert result is None
+
+    def test_decode_and_validate_jwt_invalid_signature(self):
+        """Test JWT validation with invalid signature."""
+        jwks_endpoint = "https://example.com/.well-known/jwks.json"
+        issuer = "https://example.com"
+        audience = "test-client"
+        
+        verifier = JWKSTokenVerifier(jwks_endpoint, issuer, audience)
+        
+        # Mock JWT decode to raise an exception for invalid signature
+        with patch('jwt.decode', side_effect=jwt.InvalidSignatureError("Invalid signature")):
+            result = verifier._decode_and_validate_jwt("mock.jwt.token", "mock-public-key")
+            
+            assert result is None
+
+    def test_decode_and_validate_jwt_expired_token(self):
+        """Test JWT validation with expired token."""
+        jwks_endpoint = "https://example.com/.well-known/jwks.json"
+        issuer = "https://example.com"
+        audience = "test-client"
+        
+        verifier = JWKSTokenVerifier(jwks_endpoint, issuer, audience)
+        
+        # Mock JWT decode to raise an exception for expired token
+        with patch('jwt.decode', side_effect=jwt.ExpiredSignatureError("Token has expired")):
+            result = verifier._decode_and_validate_jwt("mock.jwt.token", "mock-public-key")
+            
+            assert result is None
+
+    def test_decode_and_validate_jwt_invalid_issuer(self):
+        """Test JWT validation with invalid issuer."""
+        jwks_endpoint = "https://example.com/.well-known/jwks.json"
+        issuer = "https://example.com"
+        audience = "test-client"
+        
+        verifier = JWKSTokenVerifier(jwks_endpoint, issuer, audience)
+        
+        # Mock JWT decode to raise an exception for invalid issuer
+        with patch('jwt.decode', side_effect=jwt.InvalidIssuerError("Invalid issuer")):
+            result = verifier._decode_and_validate_jwt("mock.jwt.token", "mock-public-key")
+            
+            assert result is None
+
+    def test_decode_and_validate_jwt_invalid_audience(self):
+        """Test JWT validation with invalid audience."""
+        jwks_endpoint = "https://example.com/.well-known/jwks.json"
+        issuer = "https://example.com"
+        audience = "test-client"
+        
+        verifier = JWKSTokenVerifier(jwks_endpoint, issuer, audience)
+        
+        # Mock JWT decode to raise an exception for invalid audience
+        with patch('jwt.decode', side_effect=jwt.InvalidAudienceError("Invalid audience")):
```

Comment:
this verifies that the behavior is correct if the audience is bad, it doesn't verify that we've configured the verifier correctly though - the call to decode will *always* raise a validator exception. We should probably have a test that ensures that inputs that don't match do the right thing.